proposals/lip-0053.md

```diff
@@ -54,23 +54,22 @@ The `inboxUpdate` contains the information relative to the messages to be includ
 
 An array of cross-chain messages. See [LIP 0049][lip-0049] for the general properties and processing of cross-chain messages. The `crossChainMessages` property must contain consecutive messages from the outbox.
 
-##### messageWitness
+##### messageWitnessHashes
 
-In the sending chain, all cross-chain messages are added to a regular Merkle tree attested by the outbox root. The `messageWitness` contains two values relative to that Merkle tree:
-the `partnerChainOutboxSize`, giving the total messages sent by the partner chain, and the `siblingHashes`, which is part of a regular Merkle proof.
+In the sending chain, all cross-chain messages are added to a regular Merkle tree attested by the outbox root. The `messageWitnessHashes` are the sibling hashes necessary for the Merkle proof.
 
 This Merkle proof is required if not all messages necessary to recompute the outbox root were given in `crossChainMessages`. If all messages required to compute the outbox root are included in `crossChainMessages` then `siblingHashes` can be left empty.
 
 ##### outboxRootWitness
 
-The outbox root of the sending chain is part of a sparse Merkle tree attested by the state root (provided in the certificate). The `outboxRootWitness` property is an inclusion proof into the state root of the outbox root computed from the receiving chain inbox, `crossChainMessages` and `messageWitness`. If the cross-chain update transaction contains an empty certificate, this property can also be left empty, as the required root was already attested in a previous cross-chain update.
+The outbox root of the sending chain is part of a sparse Merkle tree attested by the state root (provided in the certificate). The `outboxRootWitness` property is an inclusion proof into the state root of the outbox root computed from the receiving chain inbox, `crossChainMessages` and `messageWitnessHashes`. If the cross-chain update transaction contains an empty certificate, this property can also be left empty, as the required root was already attested in a previous cross-chain update.
 
 ### Posting Cross-chain Updates on Mainchain and on Sidechains
 
 As the roles of both mainchain and sidechain are quite different, so are the transactions used to post cross-chain updates. The most notable differences are:
 
 * On the mainchain, all CCMs included in the `inboxUpdate` must have their `sendingChainID` equal to the chainID sending the cross-chain update.
-* On sidechains, all CCMs included in the `inboxUpdate` must have their `receivingChainID` equal to the chainID of the sidechain receiving the cross-chain update.
+* On sidechains, all CCMs included in the `inboxUpdate` must have their `receivingChainID` equal to the chainID of the sidechain receiving the cross-chain update (if has the default "ok" status).
```

Comment:
Don't we always use the `send` or `error` method, so that the receivingChainID is actually the ID of the receiving chain?

Reply:
we have to handle the case of messages bounced on the mainchain. I will clarify this better.

Reply:
that was done in the previous protocol and we decided to change it and handle it like this to avoid that there are messages sent from the mainchain where the sending chain is another chain (possibly with the same nonce of another message). Furthermore, using `error` increases the mainchain nonce, while using the wrong sendingChainID (the one of the original receiving chain)

Reply:
Can be resolved as we concluded to not have a special case for bouncing any more.